homeassistant/components/balboa/fan.py

```diff
@@ -0,0 +1,82 @@
+"""Support for Balboa Spa Pumps."""
+from __future__ import annotations
+
+from typing import Any
+
+from pybalboa import SpaClient, SpaControl
+
+from homeassistant.components.fan import FanEntity, FanEntityFeature
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util.percentage import (
+    ordered_list_item_to_percentage,
+    percentage_to_ordered_list_item,
+)
+
+from .const import DOMAIN
+from .entity import BalboaControlEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the spa's pumps as FAN entities."""
```

Comment:
I think representing these as fans is stretching the entity model a bit far.

Can these be number entities instead?

Reply:
I would argue that a pump is a fan that moves liquid instead of air and seems more intuitive than a number entity.